### PR TITLE
upgrade react-navigation to v5 and refactor navigation

### PR DESCRIPTION
**Pull request by dereckquock**
https://reactnavigation.org/docs/upgrading-from-4.x
https://trello.com/c/68V4Ngrk/129-upgrade-react-navigation-to-v5

### CODE REVIEW

**Review by Phillip3 — DISMISSED** [review]
Didn’t have time to fully look at this, but what I looked at looks good.

File: components/GroupCardDetails.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import React from 'react';
 import { StyleSheet, View, TouchableHighlight } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
```
Comment (dereckquock) [inline components/GroupCardDetails.js:5]: we can use `navigation` without prop drilling 🔥

File: navigation/AppNavigator.js [hunk 2]
```diff
@@ -1,10 +1,213 @@
...
+      />
+    </GroupsStack.Navigator>
+  );
+}
+
+const Tab = createBottomTabNavigator();
+
+export default function App() {
+  return (
+    <NavigationContainer>
```
Comment (dereckquock) [inline navigation/AppNavigator.js:142]: this is great 🔥 and super declarative 👍

File: screens/Activate.js [hunk 3]
```diff
@@ -1,66 +1,57 @@
...
-        <Text style={styles.heading}>Your next steps</Text>
-        <Text style={styles.content}>
-          Activate is about you and the environments we design at Echo.Church
-          that can lead you to discover your purpose, develop friendships, and
-          make a difference. Activate is a free dinner experience where we can
-          get to know you and give you a glimpse of what being connected at
-          Echo.Church is all about.
-        </Text>
+    <HeaderHeightContext.Consumer>
+      {headerHeight => (
```
Comment (dereckquock) [inline screens/Activate.js:13]: we can use the `headerHeight` from context to set the correct padding 🔥

File: screens/Groups.js [hunk 4]
```diff
@@ -66,22 +67,24 @@ function useQuery(groups) {
     name?.toLowerCase().includes(debouncedQuery?.toLowerCase())
   );
 
-  if (query) {
-    Amplitude.logEventWithProperties('SEARCH Groups', {
-      search_text: debouncedQuery,
-    });
-  }
+  useEffect(() => {
```
Comment (dereckquock) [inline screens/Groups.js:70]: oops 😅

File: screens/Home.js [hunk 5]
```diff
@@ -37,6 +38,12 @@ const getStoredPosts = () => {
 };
 
 const HomeScreen = () => {
+  useHandleTabChange('Home');
```
Comment (dereckquock) [inline screens/Home.js:41]: custom hook to log when we've changed focus to this tab

File: screens/Home.js [hunk 6]
```diff
@@ -37,6 +38,12 @@ const getStoredPosts = () => {
...
+  const insets = useSafeArea();
+  const ref = React.useRef(null);
+
+  useScrollToTop(ref);
```
Comment (dereckquock) [inline screens/Home.js:45]: allows us to scroll to the top when the user taps on the tab bar item 🔥

**Pull request closed without merging**